### ISSUE TITLE
Multiple node-gyp errors when starting a new solid-start project - seems to be related to node_modules/@parcel/watcher

### ISSUE THREAD

**Issue by componentscience** [solidjs/solid-start#1310]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Multiple node/gyp errors seemingly related to node_modules/@parcel/watcher when running 'npm install' in a new solid-start project. 

$ npm init solid@latest
$ cd my-app
$ npm install

Install options:
  basic
  SSR yes
  Typescript yes

Output:

[host@myhost site]$ npm install

npm ERR! code 1
npm ERR! path /home/host/site/node_modules/@parcel/watcher
npm ERR! command failed
npm ERR! command sh -c node-gyp rebuild
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@10.0.1
npm ERR! gyp info using node@20.11.0 | linux | x64
npm ERR! gyp info find Python using Python version 3.11.6 found at "/sbin/python3"
npm ERR! gyp info spawn /sbin/python3
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args '/home/host/.nvm/versions/node/v20.11.0/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args 'binding.gyp',
npm ERR! gyp info spawn args '-f',
npm ERR! gyp info spawn args 'make',
npm ERR! gyp info spawn args '-I',
npm ERR! gyp info spawn args '/home/host/site/node_modules/@parcel/watcher/build/config.gypi',
npm ERR! gyp info spawn args '-I',
npm ERR! gyp info spawn args '/home/host/.nvm/versions/node/v20.11.0/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args '-I',
npm ERR! gyp info spawn args '/home/host/.cache/node-gyp/20.11.0/include/node/common.gypi',
npm ERR! gyp info spawn args '-Dlibrary=shared_library',
npm ERR! gyp info spawn args '-Dvisibility=default',
npm ERR! gyp info spawn args '-Dnode_root_dir=/home/host/.cache/node-gyp/20.11.0',
npm ERR! gyp info spawn args '-Dnode_gyp_dir=/home/host/.nvm/versions/node/v20.11.0/lib/node_modules/npm/node_modules/node-gyp',
npm ERR! gyp info spawn args '-Dnode_lib_file=/home/host/.cache/node-gyp/20.11.0/<(target_arch)/node.lib',
npm ERR! gyp info spawn args '-Dmodule_root_dir=/home/host/site/node_modules/@parcel/watcher',
npm ERR! gyp info spawn args '-Dnode_engine=v8',
npm ERR! gyp info spawn args '--depth=.',
npm ERR! gyp info spawn args '--no-parallel',
npm ERR! gyp info spawn args '--generator-output',
npm ERR! gyp info spawn args 'build',
npm ERR! gyp info spawn args '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp: binding.gyp not found (cwd: /home/host/site/node_modules/@parcel/watcher) while trying to load binding.gyp
npm ERR! gyp ERR! configure error 
npm ERR! gyp ERR! stack Error: `gyp` failed with exit code: 1
npm ERR! gyp ERR! stack at ChildProcess.<anonymous> (/home/host/.nvm/versions/node/v20.11.0/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:271:18)
npm ERR! gyp ERR! stack at ChildProcess.emit (node:events:518:28)
npm ERR! gyp ERR! stack at ChildProcess._handle.onexit (node:internal/child_process:294:12)
npm ERR! gyp ERR! System Linux 6.7.3-arch1-1
npm ERR! gyp ERR! command "/home/host/.nvm/versions/node/v20.11.0/bin/node" "/home/host/.nvm/versions/node/v20.11.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
npm ERR! gyp ERR! cwd /home/host/site/node_modules/@parcel/watcher
npm ERR! gyp ERR! node -v v20.11.0
npm ERR! gyp ERR! node-gyp -v v10.0.1
npm ERR! gyp ERR! not ok

npm ERR! A complete log of this run can be found in: /home/host/.npm/_logs/2024-02-05T14_27_16_860Z-debug-0.log

### Expected behavior 🤔

Complete 'npm install' when creating a new solid-start project.

### Steps to reproduce 🕹

Steps:

1. $ npm init solid@latest
2. $ cd my-app
3. $ npm install

Install options:
  basic
  SSR yes
  Typescript yes

### Your environment 🌎

```shell
$ node –version
v20.11.0
$ node-gyp –version
v10.0.1
$ npm –version
10.4.0
$ uname -a
Linux 6.7.3-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 01 Feb 2024 10:30:35 +0000 x86_64 GNU/Linux
```

**Comment by lxsmnsyc** [comment]
why was there a Parcel

**Comment by componentscience** [comment]
@lxsmnsyc I'm not sure I understand the question. @parcel/watcher was referenced based on the error output. I edited the title slightly to hopefully make it more intuitive.

**Comment by darrenswhite** [comment]
adding this into `package.json` worked for me:

```
  "overrides": {
    "@parcel/watcher": "2.1.0"
  }
```

**Comment by ryansolid** [comment]
Hmm it looks like `@vinxi/listhen@1.5.6` uses `@parcel/watcher`. That being said if a version issue is happening that is upstream. I haven't seen this issue and I don't imagine it is related to our package locks. It looks like the current version we install for `@parcel/watcher` is `2.4.0`

Looking further I see certain combinations seem to be having some issues with npm/node versions. https://github.com/parcel-bundler/watcher/issues/156

I don't think there is much for us to do here. If Vinxi needs to reconsider libraries I suppose we could look that but it seems that there are some challenges with this package.

**Comment by devongovett** [comment]
Please try `@parcel/watcher` v2.4.1, which should fix this issue with npm 10.4.0. See https://github.com/parcel-bundler/watcher/issues/156#issuecomment-1961552001 for details.

**Comment by nksaraf** [comment]
Closing in favor or @devongovett's comment (https://github.com/solidjs/solid-start/issues/1310#issuecomment-1961559146). If its still an issue, reopen this.
